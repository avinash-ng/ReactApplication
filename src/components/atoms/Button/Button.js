import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
