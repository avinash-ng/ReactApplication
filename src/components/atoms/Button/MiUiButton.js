import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton