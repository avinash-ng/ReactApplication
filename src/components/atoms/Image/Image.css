img {
    width: 450px;
    height: 450px;
    float: left;
    margin-left: 20px;
    margin-top: 0px;
    padding-top: 60px;
    /* margin-bottom: 30px; */
}