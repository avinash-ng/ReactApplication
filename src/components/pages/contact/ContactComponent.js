import React, {useState} from 'react';
import {useAuth0} from '../../../react-auth0-spa'
// import './style.css'
import DrawerComponent from "../../template/pagestemplate/DrawerComponent"
// import './ContactPage.css'
import Label from "../../atoms/Label/Label"
import {makeStyles} from '@material-ui/core/styles'
import axios from 'axios';
// import Input from "../../atoms/Button/Button"

const useStyles = makeStyles((theme) => ({
    input: {
        width: '500px',
        height: '40px'
    },
    contactform: {
        border: '2px solid black',
        backgroundColor: 'silver',
        width: '600px',
        marginLeft: '500px',
        borderRadius: '10px'
    },
    button: {
        marginBottom: '20px',
        width: '100px',
        marginLeft: '0px',
        backgroundColor: 'black',
        color: 'white',
        fontSize: '20px'

    },
    div: {
        marginTop: '20px',
        marginBottom: '20px',
    },
    textarea: {
        width: '500px',
        height: '60px'
    },
    
}))

const ContentComponent = () => {
    const { user } = useAuth0() || {}
    
    const classes = useStyles();
    
    var values = ["india", "russia", "venezuela"]
    var names = ["India", "Russia", "Venezuela"]
    const [mail, updateMail] = useState()
    const mailId = user === undefined ? '16pa1a0576@vishnu.edu.in': user.mail
    var isInvalid = mailId !== mail || mail === '';

    const changeHandler = (e) => {
        updateMail(e.target.value)
    }
    const submit = () => {
        var data = {
            email: document.getElementsByClassName('fname')[0].value,
            message: document.getElementsByClassName('message')[0].value,
            password: document.getElementsByClassName('password')[0].value
        }
        axios.post('http://localhost:3001/sendmail', data).then(response => alert("mail sent successfully"))
    }
    return (
        <React.Fragment>
            <meta name="viewport" content="width=device-width, initial-scale=1"></meta>
            <div className="content-comp"> 
                <h1>Get In Touch With Us</h1>
                <h2>If you have any queries please feel free to contact us.</h2>
                <form className="contact-form" onSubmit={submit} className={classes.contactform}>
                    <div className="name-section">
                    
                        <div className="firstname" className={classes.div}>   
                            <label style={{paddingLeft: '-10px' , fontSize: '18px'}}>First Name </label> <br/>
                            <input className="fname" type="text" className={classes.input} required></input> <br/>
                        </div>

                        <div className="lastname" className={classes.div}>
                            <label style={{fontSize: '18px'}}>Last Name </label><br />
                            <input className="lname" type="text" className={classes.input} required></input><br />
                        </div>

                        <div className="email" className={classes.div}>
                            <label >Email   </label><br />
                            <input className="mail" className={classes.input} value={mail} onChange={changeHandler} type="email" required /><br />
                        </div>
                        
                        <div className="messaging-div" className={classes.div}>
                            <label style={{ fontSize: '18px' }}>leave a message</label> <br/>
                            <textarea className="message" className={classes.textarea} id="subject" name="subject" placeholder="Write something.." defaultValue="you have to write something" required></textarea><br/>
                        </div>

                        <div> 
                            <button className='button' className={classes.button} disabled={isInvalid} type="submit" >Submit</button> 
                        </div>
                    </div>
                </form>
            </div>
            
            
        </React.Fragment>
    )
}
const AboutComponent = () => {
    
    return (
        
        <div>
            <DrawerComponent > <ContentComponent /> </DrawerComponent>   
            
        </div>
    )
}
export default AboutComponent;


