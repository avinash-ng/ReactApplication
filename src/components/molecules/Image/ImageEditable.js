import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable