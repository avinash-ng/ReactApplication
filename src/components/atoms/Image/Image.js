import React from 'react'
import PropTypes from "prop-types"
import "./Image.css"
const Image = (props) => {
    return (
        <img data-testid="image" src={props.imageName} alt={props.altText} />
    )
}
Image.propTypes = {
    imageName: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
} 

export default Image