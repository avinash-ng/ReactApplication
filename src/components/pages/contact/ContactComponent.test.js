const { shallow, configure, mount } = require("enzyme")
import ContactComponent from "./ContactComponent"
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import Image from "../../atoms/Image/Image";
import ImageEditable from "../../molecules/Image/ImageEditable";
import { ListItem, AppBar } from '@material-ui/core'
import { useAuth0 } from '../../../react-auth0-spa'
import { TabPanel } from "./ContactComponent";

configure({ adapter: new Adapter() });
const user = {
    email: "av@gmail.com",
    email_verified: true,
    sub: "google-oauth2|12345678901234"
}
jest.mock("../../../react-auth0-spa");
describe('testing <ContactComponent /> ', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            user,
            logout: jest.fn(),
            loginWithRedirect: jest.fn()
        });
    });

    it('renders content-comp correctly', () => {

        const RenderComponent = mount(<ContactComponent> </ContactComponent>)
        expect(RenderComponent.find('.content-comp').exists()).toBe(true)
    })

    it('should render button correctly', () => {
        const RenderComponent = mount(<ContactComponent > </ContactComponent>)
        expect(RenderComponent.find('button').exists()).toBe(true)
    })



    it('matches snapshot', () => {
        const RenderComponent = shallow(<ContactComponent ></ContactComponent>)
        expect(RenderComponent).toMatchSnapshot()
    })
})