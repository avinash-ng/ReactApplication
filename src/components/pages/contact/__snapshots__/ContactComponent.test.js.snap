// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <ContactComponent />  matches snapshot 1`] = `ShallowWrapper {}`;
